clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -315,6 +315,19 @@ private synchronized void disableHeartbeatThread() {
             heartbeatThread.disable();
     }
 
+    private void closeHeartbeatThread() {
+        if (heartbeatThread != null) {
+            heartbeatThread.close();
+
+            try {
+                heartbeatThread.join();
+            } catch (InterruptedException e) {
+                log.warn("Interrupted while waiting for consumer heartbeat thread to close");
+                throw new InterruptException(e);
```

Comment:
My bad, I hadn't noticed that the rethrown exception was actually different (subtle name difference). I blame the small screen on my phone. ;)